clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1030,6 +1030,7 @@ public void assign(Collection<TopicPartition> partitions) {
      * @return The fetched records (may be empty)
      */
     private Map<TopicPartition, List<ConsumerRecord<K, V>>> pollOnce(long timeout) {
+        client.maybeTriggerWakeup();
```

Comment:
`hasFetchedRecords` avoids the cost of populating the map returned by `fetchRecords`. No locking is needed for that. So, if I understood the suggestion right, it would look something like:

```java
if (fetcher.hasFetchedRecords && client.hasPendingWakeup())
    client.poll(0, ...)
```

I guess by calling `poll(0, ...)`, we don't have to expose `maybeTriggerWakeup()`.